scipy/stats/_continuous_distns.py

```diff
@@ -6942,6 +6942,116 @@ def _stats(self, a):
+        if 0 < shape <= 1:
+            return shape, loc, scale
+
+        # if `shape > 1`, then we need to fit under the assumption that
+        # `shape > 1` and redefine analytical functions for this new
+        # assumption.
+
+        def location_sgt1(data, scale):
+            # analytical solution for `loc` when `shape > 1` based on the
+            # scale.
+            return data.max() - scale
+
+        # if the location or scale is fixed, an analytical solution is
+        # available.
+        if {floc, fscale} != {None}:
+            if floc is not None:
+                scale = fscale or scale_universal(data, floc)
+                return ((fshape or shape_universal(data, floc, scale)), floc,
+                        scale)
+            if fscale is not None:
+                loc = np.nextafter(location_sgt1(data, fscale), np.inf)
+                return ((fshape or shape_universal(data, loc, scale)), loc,
+                        fscale)
+
+        # The support of the distribution is `(x - loc)/scale > 0`.
+        # If all parameters are free, or only the shape is fixed, the
+        # method of Lagrange multipliers turns this constraint into an
+        # equation that can be solved numerically. 
+
+        def dL_dLocation(data, shape, scale):
+            # The partial derivative of the log-likelihood function w.r.t.
+            # the location.
+            return - data.shape[0] * shape / scale
+
+        def dL_dScale(data, shape, loc):
+            # The partial derivative of the log-likelihood function w.r.t.
+            # the scale.
+            mask = data != loc
+            return (shape - 1) * np.sum(1 / (loc - data[mask]))
+
+        def fun_to_solve(loc):
+            # optimize the location by setting the partial derivatives
+            # w.r.t. to location and scale equal and solving.
+            scale = np.nextafter(scale_universal(data, loc), -np.inf)
+            shape = fshape or shape_universal(data, loc, scale)
+            return dL_dLocation(data, shape, scale) - dL_dScale(data, shape,
+                                                                loc)
+
+        # set brackets for `root_scalar` to use when optimizing over the
+        # location such that a root is likely between them.
+        lbrack, rbrack = data.min() - 1, np.nextafter(data.min(), -np.inf)
+
+        def interval_contains_root(lbrack, rbrack):
+            # Check if the interval (lbrack, rbrack) contains the root.
+            return (np.sign(fun_to_solve(lbrack)) !=
+                    np.sign(fun_to_solve(rbrack)))
+
+        # if a root is not between the brackets, iteratively expand the left
+        # bracket until they include a sign change. The right bracket is
+        # already against the maximum permissible value for this data so
+        # only the left bracket is expanded.
+        while not interval_contains_root(lbrack, rbrack) and lbrack != -np.inf:
+            lbrack -= np.abs(lbrack)
```

Comment:
This could be why the code appears to hang. What if `data.min()` ~ 1? Then `lbrack` starts and stays at 0, right? Also, let's make the grow exponentially (e.g. double in size each time), not linearly.